contrib/mesos/pkg/scheduler/components/algorithm/podschedulers/fcfs.go

```diff
@@ -82,25 +60,44 @@ func (fps *fcfsPodScheduler) SchedulePod(r offers.Registry, task *podtask.T) (of
 			return false, nil
 		}
 
-		if fps.FitPredicate()(task, offer, n) {
-			if p.Acquire() {
-				acceptedOffer = p
-				log.V(3).Infof("Pod %s accepted offer %v", podName, offer.Id.GetValue())
-				return true, nil // stop, we found an offer
-			}
+		spec := &podtask.Spec{}
+		err := fps.procurement.Procure(task, offer, n, spec)
```

Comment:
Very good spot. We have to clone the offer here. I would have bet that we did that, but looking at the code now, we don't.